pkg/build/controller/build/build_controller.go

```diff
@@ -1583,6 +1584,17 @@ func (bc *BuildController) handleControllerConfig() error {
 		bc.additionalTrustedCAData = nil
 		return nil
 	}
+	if config.Spec.BuildDefaults.DefaultProxy != nil {
+		if config.Spec.BuildDefaults.DefaultProxy.HTTPProxy != "" {
+			bc.buildDefaults.Config.Env = append(bc.buildDefaults.Config.Env, corev1.EnvVar{Name: "HTTP_PROXY", Value: config.Spec.BuildDefaults.DefaultProxy.HTTPProxy})
+		}
+		if config.Spec.BuildDefaults.DefaultProxy.HTTPSProxy != "" {
+			bc.buildDefaults.Config.Env = append(bc.buildDefaults.Config.Env, corev1.EnvVar{Name: "HTTPS_PROXY", Value: config.Spec.BuildDefaults.DefaultProxy.HTTPSProxy})
+		}
+		if config.Spec.BuildDefaults.DefaultProxy.NoProxy != "" {
+			bc.buildDefaults.Config.Env = append(bc.buildDefaults.Config.Env, corev1.EnvVar{Name: "NO_PROXY", Value: config.Spec.BuildDefaults.DefaultProxy.NoProxy})
+		}
+	}
```

Comment:
no, sorry, but this is still wrong.  builddefaults.config.env is for adding default env vars to the user's app image.  We do not want these vars added to the app image.